### ISSUE TITLE
Segfault with returning `type` with comptime-discarded value inside

### ISSUE THREAD

**Issue by r00ster91** [ziglang/zig#13107]
### Zig Version

0.10.0-dev.4247+3234e8de3

### Steps to Reproduce

```zig
const A = struct {};

fn B(comptime a: A) type {
    return struct {
        comptime {
            _ = a;
        }
    };
}

comptime {
    B(.{});
}
```

### Expected Behavior

```
$ zig run x.zig
$
```

### Actual Behavior

```
$ zig run x.zig
Segmentation fault (core dumped)
$
```

**Comment by r00ster91** [comment]
Here are two more that I'm not sure of if they're caused by the same issue or not.
So when this is fixed, please verify that these are fixed too:
```zig
const something = function();

pub fn main() void {
    _ = something;
}

fn function() void {
    Thing(u8).function();
    return 0;
}

fn Thing(comptime T: type) type {
    return struct {
        fn function() void {
            _ = T;
        }
    };
}
```
```zig
const something = function();

pub fn main() void {
    _ = something;
}

fn function() void {
    Thing(u8);
}

fn Thing(comptime T: type) type {
    return struct {
        comptime {
            _ = T;
        }
    };
}
```
If not, I can open new issues for these.

**Comment by Vexu** [comment]
Duplicate of #12765, fixed by #13017
